matisse/src/main/java/com/zhihu/matisse/internal/loader/AlbumMediaLoader.java

```diff
@@ -73,6 +89,28 @@ private AlbumMediaLoader(Context context, Uri uri, String[] projection, String s
 
     public static CursorLoader newInstance(Context context, Album album, boolean capture) {
         if (album.isAll()) {
+            SelectionSpec selectionSpec = SelectionSpec.getInstance();
+
+            if (restrictedToImages(selectionSpec)) {
+                return new AlbumMediaLoader(
+                        context,
+                        QUERY_URI,
+                        PROJECTION,
+                        SELECTION_ALL,
+                        SELECTION_IMAGE_ARGS,
+                        ORDER_BY,
+                        capture);
+            } else if (restrictedToVideos(selectionSpec)) {
+                return new AlbumMediaLoader(
+                        context,
+                        QUERY_URI,
+                        PROJECTION,
+                        SELECTION_ALL,
+                        SELECTION_VIDEO_ARGS,
+                        ORDER_BY,
+                        capture);
+            }
+
```

Comment:
of course @gejiaheng I've made this workaround in order to supply my in-house demand but I agree with you that I'm not so much familiar with codebase, so if you could dig into this feature, I'll sure that you'll make it much better than me. :)

Sorry, I didn't realize that my old commits were going with this one...